SmartDeviceLink/SDLStreamingVideoLifecycleManager.m

```diff
@@ -550,16 +560,30 @@ - (void)sdl_handleVideoStartServiceNak:(SDLProtocolMessage *)videoStartServiceNa
     [self sdl_sendVideoStartService];
 }
 
-#pragma mark Video / Audio End Service
+#pragma mark End Service ACK/NAK
 
 - (void)handleProtocolEndServiceACKMessage:(SDLProtocolMessage *)endServiceACK {
-    SDLLogD(@"%@ service ended", (endServiceACK.header.serviceType == SDLServiceTypeVideo ? @"Video" : @"Audio"));
-    [self sdl_transitionToStoppedState:endServiceACK.header.serviceType];
+    if (endServiceACK.header.serviceType != SDLServiceTypeVideo) { return; }
+
+    SDLLogD(@"Request to end video service ACKed");
+    if (self.videoServiceEndedCompletionHandler != nil) {
+        self.videoServiceEndedCompletionHandler();
+        self.videoServiceEndedCompletionHandler = nil;
+    }
+
+    [self.videoStreamStateMachine transitionToState:SDLVideoStreamManagerStateStopped];
 }
 
 - (void)handleProtocolEndServiceNAKMessage:(SDLProtocolMessage *)endServiceNAK {
-    SDLLogW(@"%@ service ended with end service NAK", (endServiceNAK.header.serviceType == SDLServiceTypeVideo ? @"Video" : @"Audio"));
-    [self sdl_transitionToStoppedState:endServiceNAK.header.serviceType];
+    if (endServiceNAK.header.serviceType != SDLServiceTypeVideo) { return; }
+
+    SDLLogE(@"Request to end video service NAKed");
+    if (self.videoServiceEndedCompletionHandler != nil) {
+        self.videoServiceEndedCompletionHandler();
+        self.videoServiceEndedCompletionHandler = nil;
+    }
+
+    [self.videoStreamStateMachine transitionToState:SDLVideoStreamManagerStateStopped];
```

Comment:
Technically, we're not stopped if we were NAK'd. I don't know that we have a strategy for this right now.

Reply:
We will also get a `NAK` if the mobile library is streaming video but the HMI does not think we are streaming video.